homeassistant/components/honeywell/humidifier.py

```diff
@@ -0,0 +1,134 @@
+    on: Callable[[Device], Any]
+
+
+HUMIDIFIERS: dict[str, HoneywellHumidifierEntityDescription] = {
+    "Humidifier": HoneywellHumidifierEntityDescription(
+        key=HUMIDIFIER_KEY,
+        translation_key=HUMIDIFIER_KEY,
+        current_humidity=lambda device: device.current_humidity,
+        set_humidity=lambda device, humidity: device.set_humidifier_setpoint(humidity),
+        min_humidity=lambda device: device.humidifier_lower_limit,
+        max_humidity=lambda device: device.humidifier_upper_limit,
+        current_set_humidity=lambda device: device.humidifier_setpoint,
+        mode=lambda device: device.humidifier_mode,
+        off=lambda device: device.set_humidifier_off(),
+        on=lambda device: device.set_humidifier_auto(),
+        device_class=HumidifierDeviceClass.HUMIDIFIER,
+    ),
+    "Dehumidifier": HoneywellHumidifierEntityDescription(
+        key=DEHUMIDIFIER_KEY,
+        translation_key=DEHUMIDIFIER_KEY,
+        current_humidity=lambda device: device.current_humidity,
+        set_humidity=lambda device, humidity: device.set_dehumidifier_setpoint(
+            humidity
+        ),
+        min_humidity=lambda device: device.dehumidifier_lower_limit,
+        max_humidity=lambda device: device.dehumidifier_upper_limit,
+        current_set_humidity=lambda device: device.dehumidifier_setpoint,
+        mode=lambda device: device.dehumidifier_mode,
+        off=lambda device: device.set_dehumidifier_off(),
+        on=lambda device: device.set_dehumidifier_auto(),
+        device_class=HumidifierDeviceClass.DEHUMIDIFIER,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HoneywellConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Honeywell (de)humidifier dynamically."""
+    data = config_entry.runtime_data
+    entities: list = []
+    for device in data.devices.values():
+        if device.has_humidifier:
+            entities.append(HoneywellHumidifier(device, HUMIDIFIERS["Humidifier"]))
+        if device.has_dehumidifier:
+            entities.append(HoneywellHumidifier(device, HUMIDIFIERS["Dehumidifier"]))
+
+    async_add_entities(entities)
+
+
+class HoneywellHumidifier(HumidifierEntity):
+    """Representation of a Honeywell US (De)Humidifier."""
+
+    entity_description: HoneywellHumidifierEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(self, device, description) -> None:
+        """Initialize the (De)Humidifier."""
+        self._device = device
+        self.entity_description = description
+        self._attr_unique_id = f"{device.deviceid}_{description.key}"
+        self._attr_min_humidity = description.min_humidity(device)
+        self._attr_max_humidity = description.max_humidity(device)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.deviceid)},
+            name=device.name,
+            manufacturer="Honeywell",
+        )
```

Comment:
Is this a standalone device or connected?

Reply:
It's all in one "thermostat" - some have 1 output for Humidifier or Dehumidifier, some have 2 outputs for both, AND some have none :)